extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java

```diff
@@ -0,0 +1,91 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord;
+
+import com.fasterxml.jackson.databind.Module;
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import com.google.inject.multibindings.MapBinder;
+import org.apache.druid.guice.Binders;
+import org.apache.druid.guice.IndexingServiceModuleHelper;
+import org.apache.druid.guice.JsonConfigProvider;
+import org.apache.druid.guice.LazySingleton;
+import org.apache.druid.guice.PolyBind;
+import org.apache.druid.indexing.common.config.FileTaskLogsConfig;
+import org.apache.druid.indexing.common.tasklogs.FileTaskLogs;
+import org.apache.druid.indexing.overlord.TaskRunnerFactory;
+import org.apache.druid.indexing.overlord.config.TaskQueueConfig;
+import org.apache.druid.initialization.DruidModule;
+import org.apache.druid.tasklogs.NoopTaskLogs;
+import org.apache.druid.tasklogs.TaskLogKiller;
+import org.apache.druid.tasklogs.TaskLogPusher;
+import org.apache.druid.tasklogs.TaskLogs;
+
+import java.util.Collections;
+import java.util.List;
+
+public class K8sOverlordModule implements DruidModule
+{
+
+  @Override
+  public void configure(Binder binder)
+  {
+    // druid.indexer.runner.type=k8s
+    JsonConfigProvider.bind(binder, IndexingServiceModuleHelper.INDEXER_RUNNER_PROPERTY_PREFIX, KubernetesTaskRunnerConfig.class);
+    JsonConfigProvider.bind(binder, "druid.indexer.queue", TaskQueueConfig.class);
+    PolyBind.createChoice(
+        binder,
+        "druid.indexer.runner.type",
+        Key.get(TaskRunnerFactory.class),
+        Key.get(KubernetesTaskRunnerFactory.class)
+    );
+    final MapBinder<String, TaskRunnerFactory> biddy = PolyBind.optionBinder(
+        binder,
+        Key.get(TaskRunnerFactory.class)
+    );
+
+    biddy.addBinding(KubernetesTaskRunnerFactory.TYPE_NAME)
+         .to(KubernetesTaskRunnerFactory.class)
+         .in(LazySingleton.class);
+    binder.bind(KubernetesTaskRunnerFactory.class).in(LazySingleton.class);
+    configureTaskLogs(binder);
+  }
+
+  private void configureTaskLogs(Binder binder)
+  {
+    PolyBind.createChoice(binder, "druid.indexer.logs.type", Key.get(TaskLogs.class), Key.get(FileTaskLogs.class));
+    JsonConfigProvider.bind(binder, "druid.indexer.logs", FileTaskLogsConfig.class);
+
+    final MapBinder<String, TaskLogs> taskLogBinder = Binders.taskLogsBinder(binder);
+    taskLogBinder.addBinding("noop").to(NoopTaskLogs.class).in(LazySingleton.class);
+    taskLogBinder.addBinding("file").to(FileTaskLogs.class).in(LazySingleton.class);
+    binder.bind(NoopTaskLogs.class).in(LazySingleton.class);
+    binder.bind(FileTaskLogs.class).in(LazySingleton.class);
+
+    binder.bind(TaskLogPusher.class).to(TaskLogs.class);
+    binder.bind(TaskLogKiller.class).to(TaskLogs.class);
+  }
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return Collections.emptyList();
+  }
```

Comment:
Nit: the above boilerplate is no longer needed. There is a default method for the no-Jackson-modules case.

Reply:
thanks for the suggestion, I will do that.